client-tools/Shell/ProcessMonitoringFeature.cpp

```diff
@@ -49,16 +49,32 @@ void ProcessMonitoringFeature::addMonitorPID(ExternalId const& pid) {
   _monitoredProcesses.push_back(pid);
 }
 
+void ProcessMonitoringFeature::removeMonitorPIDNoLock(ExternalId const& pid) {
+  for (auto it = _monitoredProcesses.begin(); it != _monitoredProcesses.end();
+       ++it) {
+    if (it->_pid == pid._pid) {
+      _monitoredProcesses.erase(it);
```

Comment:
Please use swap with last and pop_back instead of erase from random place of of vector